File: 백준/Silver/10546. 배부른 마라토너/배부른 마라토너.py
import sys
input = sys.stdin.readline

n = int(input())
participant = [input().strip() for _ in range(n)]
didRace = [input().strip() for _ in range(n-1)]

check = {}

for p in participant:
  if not check.get(p):
    check[p] = 0
  check[p] += 1
  
for d in didRace:
  check[d] -= 1

print(list(filter(lambda x: x[1] == 1, check.items()))[0][0])

File: 백준/Silver/10546. 배부른 마라토너/배부른 마라토너.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n")

const N = Number(input[0])
const participant = input.slice(1, 1 + N)
const didRace = input.slice(1 + N)

const check = new Map()
for (let member of participant) {
  check.set(member, (check.get(member) || 0) + 1)
}
for (let member of didRace) {
  check.set(member, check.get(member) - 1)
}

for(let [key,value] of check){
    if(value){
     console.log(key)
     break
    }
}
